providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
OK, so:

1. get_restorecon_path returns the narrowest directory - this is fine. You
have a typo in comment (not forward part, backwards part).
  I know this is the nth time I'm asking, but can it be done using regex
trickery instead of checking the actual FS? I thought so but found out that
it escapes `-`, while `'a-b'[/-.*/]` works, which makes it nontrivial.
2. We use your function to determine the directory, and invoke something
like `find $BASEDIR -regex $REGEX -exec restorecon -R "{}" \;`

Good plan?

*Nitzan Raz*
http://backslasher.net

On Mon, Mar 2, 2015 at 1:27 PM, Jörg Herzinger <notifications@github.com>
wrote:

> In providers/fcontext.rb
> <https://github.com/BackSlasher/chef-selinuxpolicy/pull/3#discussion_r25591792>
> :
>
> > @@ -0,0 +1,50 @@
> > +# Support whyrun
> > +def whyrun_supported?
> > +  true
> > +end
> > +
> > +use_inline_resources
> > +
> > +# Create if doesn't exist, do not touch if fcontext is already registered
> > +action :add do
> > +  escaped_file_spec = Regexp.escape(new_resource.file_spec)
> > +  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
>
> I wouldn't execute this definition during compilation. It should be lazily
> evaluated at run time.
> Well, I looked through some of the selinux sources and I think its save to
> assume they use basic POSIX regexes.
> We could narrow the find down by using the get_restorecon_path I suggested
> above. That should definitely speed things up a lot. That should also
> remove most "access denied" problems.
>
> Giving up is not really an option in my eyes. We do have solutions, we
> just don't like them. ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BackSlasher/chef-selinuxpolicy/pull/3/files#r25591792>
> .
>